datatypes/src/raster/data_type.rs

```diff
@@ -107,6 +109,19 @@ impl RasterDataType {
             RasterDataType::F64 => true,
         }
     }
+
+    pub fn from_gdal_data_type(gdal_data_type: GDALDataType::Type) -> Result<Self> {
```

Comment:
you can replace the code in gdal.rs:221 with this function

Reply:
[d06efef](https://github.com/geo-engine/geoengine/pull/605/commits/d06efef4f2615de0a1aceabf083bc43dda51378a)